### PR TITLE
Winter fixes

### PR DESCRIPTION
**Pull request by OfekShaltiel-UTD**
Cleaned spacing, fixed cents issue in page editing, added admin functionality to page list (sees all pages), improved role based access with family concept, and added pagination to family reports table.

### CODE REVIEW

**Review by pariahGH — CHANGES_REQUESTED** [review]

File: pages/Page/[id].vue [hunk 1]
```diff
@@ -32,28 +32,52 @@ const pageData = ref<Page>({
...
-    Images: [],
+    Images: ref<Image[]>([]).value,
     status: "active",
     donation_status: "in progress",
     duration: "",
     start_date: "",
     goal_met_date: "",
-    PageDonations:[],
-    Reply:[]
+    PageDonations: ref<PageDonation[]>([]).value,
```
[the remark this answers is not in the document]
Comment (OfekShaltiel-UTD) [reply pages/Page/[id].vue:41]: I had an error when attempting to use a creation query with refs of array values being ``[]`` as part of the request body. However, I did fix it by using a destructure on the readbody(event).

File: server/api/family_page_list.get.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+const prisma = new PrismaClient()
+
+export default defineEventHandler(async event => {
+  const { cuid } = getQuery(event)
+  const result = await prisma.page.findMany({
+    where: { 
+      Family: {
+        cuid: cuid as string
+      },
+      User: {
```
[the remark this answers is not in the document]
Comment (OfekShaltiel-UTD) [reply server/api/family_page_list.get.ts:11]: I designed the page list to handle using both a family cuid and user cuid depending on several conditions.
Page list uses a user cuid / advocate from the header to load the advocates' familys' page, family cuid for families using the header, user cuid from the user list, and user cuid when an advocate creates a page (for their UX feedback of creating a page).

File: server/api/familiesReports.get.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+*	/Users
+*	function:	GET
+*	retrive details of all families from database
+*/
+
+export default defineEventHandler(async event => {
+  const { page_number } = getQuery(event)
+
+  if(event.context.user.user_role === "advocate"  || event.context.user.user_role === "admin"){
+    const [ count, families_raw, paginated_families  ] = await prisma.$transaction([
```
[the remark this answers is not in the document]
Comment (OfekShaltiel-UTD) [reply server/api/familiesReports.get.ts:17]: ``paginated_families`` is a result of a ``page.findMany`` with pagination for use in the UI table, and the CSV uses families_raw which is not paginated.

File: pages/FamilyReports.vue [hunk 4]
```diff
@@ -116,42 +128,46 @@ div
     // loads family report data from the families database table and joins and creates a download link for the file
     const loadReports = async () => {
       if( isAdminAdvocate ) { 
-          const { data: familiesData } = await useFetch('/api/families', {
-          method: 'GET' 
+          const { data: familiesData } = await useFetch('/api/familiesReports', {
```
[the remark this answers is not in the document]
Comment (OfekShaltiel-UTD) [reply pages/FamilyReports.vue:131]: Let's meet to discuss this and other usages of ``useFetch``.

File: components/CVProgress.vue [hunk 5]
```diff
@@ -1,12 +1,32 @@
...
+    } else {
+        actualBarWidth.value = 0
+        displayNothing.value = true
+    }
+
+} else if(props.modelBarWidth == "inf") {
+    actualBarWidth.value = 100
+}
+
+console.log(actualBarWidth.value)
```
[the remark this answers is not in the document]
Comment (OfekShaltiel-UTD) [reply components/CVProgress.vue:22]: alright changed this.

File: pages/FamilyReports.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (OfekShaltiel-UTD) [reply pages/FamilyReports.vue:131]: Let's make inappropriate usages of useFetch to be replaced by $fetch as a task for someone one the team to do.

File: server/api/family_page_list.get.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (OfekShaltiel-UTD) [reply server/api/family_page_list.get.ts:11]: Let's meet to redesign the page list.

File: server/api/page.post.ts [hunk 6]
```diff
@@ -11,45 +11,45 @@ const prisma = new PrismaClient()
...
-            cuid: familyCuid
-          }
-        }
+      // Creates a new entry in the database in the page model to a specfic user
+      const queryRes = await prisma.page.create({
+        data: {
+          ...data, cuid: undefined,
+          User: {
+            connect: {
+              cuid : userCuid || "0"
```
[the remark this answers is not in the document]
Comment (OfekShaltiel-UTD) [reply server/api/page.post.ts:27]: Alright changed this.

File: pages/Page/[id].vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (OfekShaltiel-UTD) [reply pages/Page/[id].vue:41]: Alright fixed. I did not have any issues with using ``[]`` for these spots when testing the backend in these spots.

**Pull request closed without merging**